core/dnsserver/onstartup.go

```diff
@@ -2,15 +2,23 @@ package dnsserver
 
 import (
 	"fmt"
+	"regexp"
 	"sort"
+
+	"github.com/coredns/coredns/plugin/pkg/dnsutil"
 )
 
+// checkZoneSyntax() checks whether the given string match 1035 Preferred Syntax or not.
+func checkZoneSyntax(zone string) bool {
+	regex1035PreferredSyntax, _ := regexp.MatchString(`^([A-Za-z][A-Za-z0-9-]*[A-Za-z](\.[A-Za-z][A-Za-z0-9-]*[A-Za-z])*\.)$`, zone)
```

Comment:
This regex looks more complex than it needs to be - and also looks wrong (doesn't match preferred 1035 syntax).  See the example regex I provided earlier.

Also, the regex match should not be executed if the zone is root or a reverse zone (since you can return true in those cases without needing to evaluate the regex).

Reply:
I think you are right here, I need to improve Regex also for trailing digits as per 1035 syntax.

Reply:
the regex match should not being executed if the zone is root or a reverse zone.